spring-cloud-context/src/main/java/org/springframework/cloud/bootstrap/BootstrapApplicationListener.java

```diff
@@ -294,6 +294,13 @@ private void apply(ConfigurableApplicationContext context, SpringApplication app
 		target.addAll(getOrderedBeansOfType(context, ApplicationContextInitializer.class));
 		application.setInitializers(target);
 		addBootstrapDecryptInitializer(application);
+
+		// Get the active profiles from the bootstrap context and set them in main
+		// application
+		// environment. This allows any profiles activated during bootstrap to be
+		// activated when
+		// config data runs in the main application context.
+		environment.setActiveProfiles(context.getEnvironment().getActiveProfiles());
```

Comment:
@ryanjbaxter  This may not fully resolve all issues mentioned in the original issue. As noted, when the spring.profiles.active configuration is sourced from a config server, it does not take effect in the main application context. The root cause, which needs fixing here, is similar: setting spring.profiles.active before the main application context is ready. However, there are other configurations besides spring.profiles.active that also need to be set in advance, such as those starting with spring.main.
I lean towards utilizing BootstrapApplicationListener to proactively locate all remote configurations for the main application context, similar to what ConfigDataLoader does.